source/framework/core/src/TRestMetadata.cxx

```diff
@@ -2255,16 +2255,38 @@ Int_t TRestMetadata::Write(const char* name, Int_t option, Int_t bufsize) {
 /// In the previous code "fPar0" is linked to "par0".
 ///
 /// Note that parameters include <parameter section and all the attributes in fElement.
+///
 void TRestMetadata::ReadAllParameters() {
     // we shall first add all the parameters to a temporary map to avoid
     // first parameter being overriden by the repeated parameter section
-    map<TString, TString> parameters;
+    map<string, string> parameters = GetParametersList();
+
+    ReadParametersList(parameters);
+}
+
+///////////////////////////////////////////////
+/// \brief It reads a parameter list and associates it to its corresponding metadata
+/// member. par0 --> fPar0.
+///
+void TRestMetadata::ReadParametersList(std::map<string, string> list) {
```

Comment:
Perhaps would be better ot pass list as argument:
void TRestMetadata::ReadParametersList(std::map<string, string> &list)

Reply:
Ok, I did it at commit b2163ac I guess we still need to follow that approach at more places ...